File: backend/Models/ProductionPlan.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace backend.Models
{
    public enum ProductionPlanStatus
    {
        Planned,
        InProgress,
        Completed,
        OnHold,
        Cancelled
    }
    public class ProductionPlan
    {
        [Key]
        public int PlanId { get; set; }

        public string ProductName { get; set; } = string.Empty;

        public int TargetQuantity { get; set; }

        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }

        public ProductionPlanStatus Status { get; set; }

        public int CreatedBy { get; set; }

        public User? Creator { get; set; }

        public ICollection<ProductionOrder>? ProductionOrders { get; set; }
    }
}

File: MPMS-Frontend/src/app/models/production-plan.ts
import { ProductionPlanStatus } from './enum';

export interface ProductionPlan {
  planId: number;
  productName: string;
  targetQuantity: number;
  startDate: string;
  endDate: string;
  status: ProductionPlanStatus;
  createdBy: number;
}

export interface ProductionPlanRequest {
  productName: string;
  targetQuantity: number;
  startDate: string;
  endDate: string;
  status: ProductionPlanStatus;
}
